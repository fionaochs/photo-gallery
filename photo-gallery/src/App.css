@import url('https://fonts.googleapis.com/css?family=Heebo|Oswald');

body {
    font-family: 'Heebo', "Segoe UI", Candara, "Bitstream Vera Sans", "DejaVu Sans", "Bitstream Vera Sans", "Trebuchet MS", Verdana, "Verdana Ref", sans-serif;
    background: rgb(65, 34, 205);
    margin: 0;
    line-height: 1.5em;
}

h1 {
    font-size: 2em;
}

h1, h2 {
    font-family: 'Oswald';
    font-weight: bold;
}
* {
    box-sizing: border-box;
}

header, main {
    padding: 10px;
}

header {
    display: flex;
    align-items: center;
    justify-content: space-around;
    background: rgb(144, 175, 242);
    border-bottom: 1px solid teal;
    margin-top: 20px;
}

h1 {
    margin: 0 20px;
    flex-grow: 1;
}

.options {
    padding: 10px 0;
}

section {
    margin: 20px auto;
    width: 90%;
    min-width: 650px;
}


.creatures {
    display: flex;
    flex-wrap: wrap;
}

.horned-types{
  /* display: flex; */
  align-items: center;
}
.creatures li {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    text-align: center;
    height: 350px;
    width: 200px;
    min-width: 200px;
    border: 1px solid #aaa;
    background-color: rgb(144, 175, 242);
    margin-top: 0px;
    margin-right: 10px;
    margin-bottom: 10px;
    margin-left: 0px;
    padding: 10px;
}

.creatures li h2 {
    font-size: 1.5em;
}

.creatures li .creature-type {
    font-size: .8em;
    color: rgb(62, 57, 126);
}

.creatures li .year {
    font-size: .8em;
}

.image-container {
    height: 50%;
}

.image-container img {
    height: 100%;
    object-fit: contain;
    width: 80%;
}

.creature-type-filter {
    height: 48px;
    width: 150px;
    font-size: 1.2em;
}