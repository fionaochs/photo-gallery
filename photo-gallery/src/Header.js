import React, { Component } from 'react';

export default class Header extends Component {
    render() {
      return <header>
          <div>
              <h1>
                  The Mythical World of Horned Creatures
              </h1>
          </div>

      </header>;
    }
}