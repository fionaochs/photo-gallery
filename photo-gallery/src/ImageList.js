import React, { Component } from 'react'

export default class CreatureList extends Component {
    render () {
        return <section className="options">

        <section className="list-section">
            <ul className="creatures">{this.props.creaturesNodes}</ul>
        </section>
            {/* creaturesNodes is the selected creature <li>
            inserts each creature <li> into <ul> holding all of them */}

        </section>
    }
}
