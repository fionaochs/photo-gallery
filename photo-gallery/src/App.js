import React, { Component } from 'react';
import images from './data.js';
import './App.css';
import Creature from './ImageItem.js';
import Header from './Header.js';
import ImageList from './ImageList.js';


class App extends Component {
    state = {
      selected: null
    };
    handleChange = e => {
      this.setState({selected: e.target.value});
    }
    render () {

      const { selected } = this.state;
      const creaturesNodes = images
      .filter(Creature => {
        if (!selected) return true;
        return Creature.keyword === selected;
      })
      .map(hornedCreature => 
        <Creature
        creature={hornedCreature} 
        />)
        
        
        return (
          <div className="App">
      <Header></Header>
      <select className="horned-types" onChange={this.handleChange}>
                <option value="" defaultValue>
                    All Horned Types
                </option>
                <option value="narwhal">Narwhal</option>
                <option value="rhino">Rhino</option>
                <option value="unicorn">Unicorn</option>
                <option value="unilego">Unilego</option>
                <option value="triceratops">Triceratops</option>
                <option value="markhor">Markhor</option>
                <option value="mouflon">Mouflon</option>
                <option value="addax">Addax</option>
                <option value="chameleon">Chameleon</option>
                <option value="lizard">Lizard</option>
                <option value="dragon">Dragon</option>
            </select>
      <ImageList creaturesNodes={creaturesNodes}></ImageList>
      </div>
      );
    }      
}
export default App;

