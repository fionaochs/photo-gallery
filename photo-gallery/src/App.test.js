import React from "react";
import App from "./App.js";
import renderer from "react-test-renderer";
import ImageList from './ImageList.js';
import Header from './Header.js';


test('renders App.js correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders Header.js correctly', () => {
  const tree = renderer
    .create(<Header />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders ImageList.js correctly', () => {
  const tree = renderer
    .create(<ImageList />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

