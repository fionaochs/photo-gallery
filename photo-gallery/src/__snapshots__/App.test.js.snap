// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App.js correctly 1`] = `
<div
  className="App"
>
  <header>
    <div>
      <h1>
        The World of Horned Creatures
      </h1>
    </div>
  </header>
  <select
    className="horned-types"
    onChange={[Function]}
  >
    <option
      defaultValue={true}
      value=""
    >
      All Horned Types
    </option>
    <option
      value="narwhal"
    >
      Narwhal
    </option>
    <option
      value="rhino"
    >
      Rhino
    </option>
    <option
      value="unicorn"
    >
      Unicorn
    </option>
    <option
      value="unilego"
    >
      Unilego
    </option>
    <option
      value="triceratops"
    >
      Triceratops
    </option>
    <option
      value="markhor"
    >
      Markhor
    </option>
    <option
      value="mouflon"
    >
      Mouflon
    </option>
    <option
      value="addax"
    >
      Addax
    </option>
    <option
      value="chameleon"
    >
      Chameleon
    </option>
    <option
      value="lizard"
    >
      Lizard
    </option>
    <option
      value="dragon"
    >
      Dragon
    </option>
  </select>
  <section
    className="options"
  >
    <section
      className="list-section"
    >
      <ul
        className="creatures"
      >
        <li>
          <div
            id="info-container"
          >
            <h2>
              UniWhal
            </h2>
            <p
              className="creature-type"
            >
              A unicorn and a narwhal nuzzling their horns
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg"
              src="http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg"
            />
          </div>
          <p
            className="horns"
          >
            1
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Rhino Family
            </h2>
            <p
              className="creature-type"
            >
              Mother (or father) rhino with two babies
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://images.unsplash.com/photo-1512636618879-bbe79107e9e3?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=bd9460ee6d1ddbb6b1ca7be86dfc4590&auto=format&fit=crop&w=1825&q=80"
              src="https://images.unsplash.com/photo-1512636618879-bbe79107e9e3?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=bd9460ee6d1ddbb6b1ca7be86dfc4590&auto=format&fit=crop&w=1825&q=80"
            />
          </div>
          <p
            className="horns"
          >
            2
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Unicorn Head
            </h2>
            <p
              className="creature-type"
            >
              Someone wearing a creepy unicorn head mask
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://www.dhresource.com/0x0s/f2-albu-g5-M00-1A-11-rBVaI1hsIIiALxKzAAIHjSU3VkE490.jpg/wholesale-halloween-costume-prop-unicorn.jpg"
              src="https://www.dhresource.com/0x0s/f2-albu-g5-M00-1A-11-rBVaI1hsIIiALxKzAAIHjSU3VkE490.jpg/wholesale-halloween-costume-prop-unicorn.jpg"
            />
          </div>
          <p
            className="horns"
          >
            1
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              UniLego
            </h2>
            <p
              className="creature-type"
            >
              Lego figurine dressed in a unicorn outfit
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://images.unsplash.com/photo-1518946222227-364f22132616?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=4836a6fca62e7dce9324346bacfde085&auto=format&fit=crop&w=2534&q=80"
              src="https://images.unsplash.com/photo-1518946222227-364f22132616?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=4836a6fca62e7dce9324346bacfde085&auto=format&fit=crop&w=2534&q=80"
            />
          </div>
          <p
            className="horns"
          >
            1
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Basically a unicorn
            </h2>
            <p
              className="creature-type"
            >
              A narwhal is basically a unicorn after all, right?
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://i.pinimg.com/736x/b4/61/06/b46106830b841017ea59870b27ec18dc--narwhals-a-unicorn.jpg"
              src="https://i.pinimg.com/736x/b4/61/06/b46106830b841017ea59870b27ec18dc--narwhals-a-unicorn.jpg"
            />
          </div>
          <p
            className="horns"
          >
            1
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              #truth
            </h2>
            <p
              className="creature-type"
            >
              The truth behind narwhals
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://i.pinimg.com/originals/16/cf/2a/16cf2a0b3fd51b9bee08bb6296193b75.jpg"
              src="https://i.pinimg.com/originals/16/cf/2a/16cf2a0b3fd51b9bee08bb6296193b75.jpg"
            />
          </div>
          <p
            className="horns"
          >
            1
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Baby Rhino
            </h2>
            <p
              className="creature-type"
            >
              This is actually a figurine but it looks kinda real
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://secure.img1-ag.wfcdn.com/im/17007094/resize-h800%5Ecompr-r85/3589/35892451/Baby+Rhino+Figurine.jpg"
              src="https://secure.img1-ag.wfcdn.com/im/17007094/resize-h800%5Ecompr-r85/3589/35892451/Baby+Rhino+Figurine.jpg"
            />
          </div>
          <p
            className="horns"
          >
            2
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Cera
            </h2>
            <p
              className="creature-type"
            >
              Three horns but still, horns. And who doesn't like The Land Before Time?
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://vignette.wikia.nocookie.net/landbeforetime/images/c/c3/Cera_infobox.png/revision/latest?cb=20180422005003"
              src="https://vignette.wikia.nocookie.net/landbeforetime/images/c/c3/Cera_infobox.png/revision/latest?cb=20180422005003"
            />
          </div>
          <p
            className="horns"
          >
            3
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Narwhal costume
            </h2>
            <p
              className="creature-type"
            >
              A woman wearing a blue narwhal costume
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://ae01.alicdn.com/kf/HTB18GwSQVXXXXaZaXXXq6xXFXXXh/Animal-Cosplay-Costume-Narwhal-Onesie-Mens-Womens-Cartoon-Whale-Pajamas.jpg"
              src="https://ae01.alicdn.com/kf/HTB18GwSQVXXXXaZaXXXq6xXFXXXh/Animal-Cosplay-Costume-Narwhal-Onesie-Mens-Womens-Cartoon-Whale-Pajamas.jpg"
            />
          </div>
          <p
            className="horns"
          >
            1
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Rhino costume
            </h2>
            <p
              className="creature-type"
            >
              Mascots have to get their costumes somewhere
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://www.shopmascot.com/image/cache/mascotnew/new196-800x800.jpg"
              src="https://www.shopmascot.com/image/cache/mascotnew/new196-800x800.jpg"
            />
          </div>
          <p
            className="horns"
          >
            2
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Believe
            </h2>
            <p
              className="creature-type"
            >
              I believe in unicorns, do you?
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://www.tinselbox.com/wp-content/uploads/2018/03/I-BELIEVE-IN-UNICORNS-FREE-PRINTABLE-WATERCOLOR-7-of-1.jpg"
              src="https://www.tinselbox.com/wp-content/uploads/2018/03/I-BELIEVE-IN-UNICORNS-FREE-PRINTABLE-WATERCOLOR-7-of-1.jpg"
            />
          </div>
          <p
            className="horns"
          >
            1
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Markhor
            </h2>
            <p
              className="creature-type"
            >
              These wild goats eat snakes, then secrete a foam that locals fight over for the antivenom properties -- how cool is that?
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://upload.wikimedia.org/wikipedia/commons/thumb/3/38/Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg/220px-Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/38/Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg/220px-Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg"
            />
          </div>
          <p
            className="horns"
          >
            2
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Baby markhor
            </h2>
            <p
              className="creature-type"
            >
              Even the babies are adorable
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="http://www.zooborns.com/.a/6a010535647bf3970b0223c84d5959200c-800wi"
              src="http://www.zooborns.com/.a/6a010535647bf3970b0223c84d5959200c-800wi"
            />
          </div>
          <p
            className="horns"
          >
            2
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Mouflon
            </h2>
            <p
              className="creature-type"
            >
              Those horns though
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://cdn.modernfarmer.com/wp-content/uploads/2014/08/28476658_9c97f35096_o.jpg"
              src="https://cdn.modernfarmer.com/wp-content/uploads/2014/08/28476658_9c97f35096_o.jpg"
            />
          </div>
          <p
            className="horns"
          >
            2
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Addax
            </h2>
            <p
              className="creature-type"
            >
              This guy is basically extinct but survives well in captivity, so they're frequently found in zoos
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://cdn.modernfarmer.com/wp-content/uploads/2014/08/addax.jpg"
              src="https://cdn.modernfarmer.com/wp-content/uploads/2014/08/addax.jpg"
            />
          </div>
          <p
            className="horns"
          >
            2
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Baby mouflon
            </h2>
            <p
              className="creature-type"
            >
              The cuteness that is a baby mouflon asleep
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://cbsnews3.cbsistatic.com/hub/i/r/2013/03/05/5b414225-a645-11e2-a3f0-029118418759/thumbnail/620x350/2d4cf24685b45c22912e64d2004fec8d/Baby_Mouflon_Wild_Sheep.jpg"
              src="https://cbsnews3.cbsistatic.com/hub/i/r/2013/03/05/5b414225-a645-11e2-a3f0-029118418759/thumbnail/620x350/2d4cf24685b45c22912e64d2004fec8d/Baby_Mouflon_Wild_Sheep.jpg"
            />
          </div>
          <p
            className="horns"
          >
            2
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Happy Jackson's Chameleon
            </h2>
            <p
              className="creature-type"
            >
              These are really common in Hawaii
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://cdn.modernfarmer.com/wp-content/uploads/2014/08/chameleon.jpg"
              src="https://cdn.modernfarmer.com/wp-content/uploads/2014/08/chameleon.jpg"
            />
          </div>
          <p
            className="horns"
          >
            2
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Serious Jackson's Chameleon
            </h2>
            <p
              className="creature-type"
            >
              This one is very serious.
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://imgc.allpostersimages.com/img/print/posters/dlillc-jackson-s-chameleon_a-G-13448768-14258384.jpg"
              src="https://imgc.allpostersimages.com/img/print/posters/dlillc-jackson-s-chameleon_a-G-13448768-14258384.jpg"
            />
          </div>
          <p
            className="horns"
          >
            3
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Horned Lizard
            </h2>
            <p
              className="creature-type"
            >
              Fave food: ants
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://www.nps.gov/band/learn/nature/images/short-horned-lizard-open-mouth-18.jpg?maxwidth=650&autorotate=false"
              src="https://www.nps.gov/band/learn/nature/images/short-horned-lizard-open-mouth-18.jpg?maxwidth=650&autorotate=false"
            />
          </div>
          <p
            className="horns"
          >
            100
          </p>
        </li>
        <li>
          <div
            id="info-container"
          >
            <h2>
              Smaug
            </h2>
            <p
              className="creature-type"
            >
              Fan illustration of Smaug from 'The Hobbit'
            </p>
          </div>
          <div
            className="image-container"
          >
            <img
              alt="https://upload.wikimedia.org/wikipedia/commons/thumb/d/df/Smaug_par_David_Demaret.jpg/290px-Smaug_par_David_Demaret.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/df/Smaug_par_David_Demaret.jpg/290px-Smaug_par_David_Demaret.jpg"
            />
          </div>
          <p
            className="horns"
          >
            100
          </p>
        </li>
      </ul>
    </section>
  </section>
</div>
`;

exports[`renders Header.js correctly 1`] = `
<header>
  <div>
    <h1>
      The World of Horned Creatures
    </h1>
  </div>
</header>
`;

exports[`renders ImageItem.js correctly 1`] = `
<section
  className="options"
>
  <section
    className="list-section"
  >
    <ul
      className="creatures"
    />
  </section>
</section>
`;

exports[`renders ImageList.js correctly 1`] = `
<section
  className="options"
>
  <section
    className="list-section"
  >
    <ul
      className="creatures"
    />
  </section>
</section>
`;
